db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLTime.java

```diff
@@ -1180,7 +1180,7 @@ public void read(Row row, int ordinal) throws StandardException {
 			setToNull();
 		else {
 			isNull = false;
-			encodedTime = computeEncodedTime(row.getDate(ordinal));
+			encodedTime = computeEncodedTime(row.getTimestamp(ordinal));
```

Comment:
This looks weird: we really get a timestamp from the ordinal so that we can get a time value? Not really sure if it's a bug or not, but it looks weird.